src/main/java/de/zalando/aruha/nakadi/controller/SubscriptionController.java

```diff
@@ -55,52 +58,69 @@ public SubscriptionController(final SubscriptionDbRepository subscriptionReposit
-                    eventTypeRepository.findByName(etName);
-                } catch (NoSuchEventTypeException e) {
-                    noneExistingEventTypes.add(etName);
-                }
-            }
-            if (!noneExistingEventTypes.isEmpty()) {
-                final String errorMessage = "Failed to create subscription, event type(s) not found: '" +
-                        StringUtils.join(noneExistingEventTypes, "','") + "'";
-                LOG.debug(errorMessage);
-                return create(UNPROCESSABLE_ENTITY, errorMessage, nativeWebRequest);
-            }
-
-            // generate subscription id and try to create subscription in DB
-            final Subscription subscription = subscriptionRepository.createSubscription(subscriptionBase);
-            return new ResponseEntity<>(subscription, HttpStatus.CREATED);
-
+            return createSubscription(subscriptionBase, request);
         } catch (final DuplicatedSubscriptionException e) {
             try {
-                // if the subscription with such parameters already exists - return it instead of creating a new one
-                final Subscription existingSubscription = subscriptionRepository.getSubscription(
-                        subscriptionBase.getOwningApplication(), subscriptionBase.getEventTypes(),
-                        subscriptionBase.getConsumerGroup());
-                return new ResponseEntity<>(existingSubscription, HttpStatus.OK);
-
+                return new ResponseEntity<>(getExistingSubscription(subscriptionBase), HttpStatus.OK);
             } catch (final NakadiException ex) {
                 LOG.error("Error occurred during fetching existing subscription", ex);
-                return create(INTERNAL_SERVER_ERROR, ex.getProblemMessage(), nativeWebRequest);
+                return create(INTERNAL_SERVER_ERROR, ex.getProblemMessage(), request);
             }
         } catch (final InternalNakadiException e) {
             LOG.error("Error occurred during subscription creation", e);
-            return create(e.asProblem(), nativeWebRequest);
+            return create(e.asProblem(), request);
+        }
+    }
+
+    private ResponseEntity<?> createSubscription(final SubscriptionBase subscriptionBase, final NativeWebRequest request)
+            throws InternalNakadiException, DuplicatedSubscriptionException {
+        final List<String> noneExistingEventTypes = checkExistingEventTypes(subscriptionBase);
+        if (!noneExistingEventTypes.isEmpty()) {
+            final String errorMessage = createErrorMessage(noneExistingEventTypes);
+            LOG.debug(errorMessage);
+            return create(UNPROCESSABLE_ENTITY, errorMessage, request);
         }
+
+        // generate subscription id and try to create subscription in DB
+        final Subscription subscription = subscriptionRepository.createSubscription(subscriptionBase);
+        return new ResponseEntity<>(subscription, HttpStatus.CREATED);
```

Comment:
I think it will be more readable. In general, I refactored the method a little bit but I have not touched this things.